ui/src/app/cohort-review/detail-tab-table/detail-tab-table.component.tsx

```diff
@@ -0,0 +1,290 @@
+import {DataTable} from 'primereact/datatable';
+import * as React from 'react';
+
+const css = `
+  body .p-datatable .p-sortable-column:not(.p-highlight):hover,
+  body .p-datatable .p-sortable-column.p-highlight {
+    color: #333333;
+    background-color: #f4f4f4;
+  }
+  body .p-datatable .p-datatable-thead > tr > th {
+    padding: 10px 5px 10px 10px;
+    vertical-align: middle;
+    border: 0;
+    border-bottom: 1px solid #c8c8c8;
+    border-left: 1px solid #c8c8c8;
+  }
+  body .p-datatable .p-datatable-thead > tr > th:first-of-type {
+    border-left: 0;
+  }
+  body .p-datatable .p-column-title {
+    display: flex;
+  }
+  .pi.pi-sort,
+  .pi.pi-sort-up,
+  .pi.pi-sort-down {
+    display: none;
+  }
+  .p-datatable .p-datatable-scrollable-wrapper {
+    border: 1px solid #c8c8c8;
+  }
+  .p-datatable .p-paginator.p-paginator-bottom {
+    border: 0;
+    margin-top: 20px;
+    background: none;
+    font-size: 12px;
+    text-align: right;
+  }
+  body .p-paginator .p-paginator-prev,
+  body .p-paginator .p-paginator-next,
+  body .p-paginator .p-paginator-pages .p-paginator-page {
+    border: 1px solid #cccccc;
+    border-radius: 3px;
+    background: #fafafa;
+    color: #2691D0;
+    height: auto;
+    width: auto;
+    min-width: 0;
+    padding: 7px;
+    margin: 0 2px;
+    line-height: 0.5rem;
+  }
+  body .p-paginator .p-paginator-prev,
+  body .p-paginator .p-paginator-next {
+    height: 28px;
+    width: 24px;
+  }
+  body .p-paginator .p-paginator-pages .p-paginator-page:focus {
+    box-shadow: 0;
+  }
+  body .p-paginator .p-paginator-pages .p-paginator-page.p-highlight {
+    background: #fafafa;
+    color: rgba(0, 0, 0, .5);
+  }
+  `;
+
+const styles = reactStyles({
+  table: {
+    fontSize: '12px',
+  },
+  tableBody: {
+    textAlign: 'left',
+    lineHeight: '0.6rem'
+  },
+  columnHeader: {
+    background: '#f4f4f4',
+    color: '#262262',
+    fontWeight: 600,
+  },
+  columnBody: {
+    background: '#ffffff',
+    padding: '5px',
+    verticalAlign: 'top',
+    textAlign: 'left',
+    borderLeft: 0,
+    borderRight: 0,
+    lineHeight: '0.6rem'
+  },
+  filterIcon: {
+    color: '#0086C1',
+    fontSize: '0.5rem',
+    float: 'right'
+  },
+  sortIcon: {
+    color: '#2691D0',
+    fontSize: '0.4rem'
+  }
+});
+
+export interface DetailTabTableProps {
+  tabname: string;
+  columns: Array<any>;
+  domain: string;
+  filterType: PageFilterType;
+  participantId: number;
+  workspace: WorkspaceData;
+}
+
+export interface DetailTabTableState {
+  data: Array<any>;
+  loading: boolean;
+  start: number;
+  rows: number;
+  sortField: string;
+  sortOrder: number;
+}
+
+export const DetailTabTable = withCurrentWorkspace()(
+  class extends React.Component<DetailTabTableProps, DetailTabTableState> {
+    dt: any;
+    constructor(props: DetailTabTableProps) {
+      super(props);
+      this.state = {
+        data: null,
+        loading: true,
+        start: 0,
+        rows: 25,
```

Comment:
`rows` doesn't seem to change anywhere, so it could be placed in a constant rather than in state.